back/routes/adventure.js

```diff
@@ -46,9 +46,23 @@ function formatRestaurant(restaurant) {
 }
 
 // ----- Get back4app restaurants -----
-async function getDatabaseRestaurants() {
-  const experiences = await new Parse.Query('Experience').find();
-  return experiences.map((exp) => exp.toJSON());
+async function getDatabaseRestaurants(distance, userLat, userLng) {
+  const allExperiences = await new Parse.Query('Experience').find();
+  const experiences = [];
+  for (const exp of allExperiences) {
+    const experience = exp.toJSON()
+    var config = {
+      method: 'get',
+      url: encodeURI(`https://maps.googleapis.com/maps/api/distancematrix/json?origins=${userLat},${userLng}&destinations=${experience.lat},${experience.lng}&key=${googleKey}`),
+      headers: { }
+    };
+    await axios(config).then((response) => {
+      if (response.data.rows[0].elements[0].distance.value <= distance) {
+        experiences.push({...experience, distance: response.data.rows[0].elements[0].distance.value})
+      } 
+    })
+  }
+  return (experiences);
 }
```

Comment:
I am thinking about maybe having an array of Promises and pushing every Promise in the array, so that afterwards I can do something like a Promise.all on that array and dealing with that result -> if statement 

Thanks for pointing this out!

Reply:
I think that sounds like a good idea! You might also think about using the `.map` function instead of pushing